add/update permalinks to not use .html

therewasaguy opened the pull request:
- remove .html extension from events and posts permalinks
- add permalinks for header nav links

therewasaguy commented:
@khoeger the files still have .html extensions, but the links that we use shouldn't. Let me know if you see a spot where a link includes ".html".

Both of these links already work, and this PR would just change the permalink so that, throughout the site, we only link to the first one
https://musichackathon.github.io/mmh_jekyll/2017-06-10
https://musichackathon.github.io/mmh_jekyll/2017-06-10.html

khoeger commented:
I checked out the branch permalinks, but still got http://localhost:4000/2017-06-10.html. [](url)

therewasaguy commented:
Ah, thanks @khoeger good catch. Just updated to use what Jekyll calls "pretty urls" for events and posts. This makes a folder with index.html inside of it. This changes relative paths so I updated `_layouts/event.html` to use absolute path for images

trevorknight commented on _layouts/event.html:
```diff
@@ -8,7 +8,7 @@ <h1 class="post-title">{{ page.title | escape }}</h1>
 
   {% if page.image %}
-    <img src="assets/events/{{page.image}}"/>
+    <img src="/assets/events/{{page.image}}"/>
```
This will break when hosted by github. 

Please use {{ site.github.url }}/assets

trevorknight commented on blog.md:
```diff
@@ -1,5 +1,6 @@
 ---
 title: Blog
 layout: blog
+permalink: blog
```
I know this is my fault, not yours, but as long as you're editing this front matter, would be willing to make the pages have the fields in the same order?
